### PR TITLE
Add operator<< to many classes

### LINKED ISSUE #2391: The classes in geometry need an operator<<
This came up while debugging #2389.

### PR DESCRIPTION
**Pull request by pleroy**
Fix #2391.

### CODE REVIEW

File: geometry/identity_body.hpp [hunk 1]
```diff
@@ -99,6 +99,12 @@ Identity<FromFrame, ToFrame> operator*(
   return Identity<FromFrame, ToFrame>();
 }
 
+template<typename FromFrame, typename ToFrame>
+std::ostream& operator<<(std::ostream& out,
+                         Identity<FromFrame, ToFrame> const& identity) {
+  return out << "𝟙";
```
Comment (eggrobin) [inline geometry/identity_body.hpp]: Shouldn't that be u8"𝟙"?

File: geometry/quaternion_body.hpp [hunk 2]
```diff
@@ -158,10 +160,21 @@ inline Quaternion Normalize(Quaternion const& quaternion) {
...
+  auto const unsigned_debug_string = [](double const d) {
+    std::string const s = DebugString(Abs(d));
+    CHECK_EQ('+', s[0]);
+    return s.substr(1);
+  };
+
+  return out << w << " "
+             << Sign(x) << " " << unsigned_debug_string(x) << " i "
+             << Sign(y) << " " << unsigned_debug_string(y) << " j "
+             << Sign(z) << " " << unsigned_debug_string(z) << " k";
```
Comment (eggrobin) [inline geometry/quaternion_body.hpp]: Fancy!

File: geometry/identity_body.hpp (hunk 1 above)
Comment (pleroy) [reply geometry/identity_body.hpp]: Done.

File: geometry/quaternion_body.hpp (hunk 2 above)
Comment (pleroy) [reply geometry/quaternion_body.hpp]: Yeah, I started to hate the `+ -`.

**Pull request merged**